app/services/idv/phone_step.rb

```diff
@@ -23,6 +16,38 @@ def failure_reason
       return :warning if idv_result[:success] != true
     end
 
+    def async_state
+      dcs_uuid = idv_session.idv_phone_step_document_capture_session_uuid
+      dcs = DocumentCaptureSession.find_by(uuid: dcs_uuid)
+      return ProofingDocumentCaptureSessionResult.none if dcs_uuid.nil?
+      return ProofingDocumentCaptureSessionResult.timed_out if dcs.nil?
+
+      proofing_job_result = dcs.load_proofing_result
+      return ProofingDocumentCaptureSessionResult.timed_out if proofing_job_result.nil?
+
+      if proofing_job_result.result
+        proofing_job_result.done!
+
+        proofing_job_result
```

Comment:
this bit about marking it as `done!` kind of threw me for a bit

- the `!` made me think it was similar to ActiveRecord methods that persist the result...but it does not
- structs are great for immutable copies but we're mutating here

I have two ideas:

1. another static initializer (I think this is my preference)

    ```ruby
    ProofingDocumentCaptureSessionResult.done(from: proofing_result_job)
    ```
    - Makes it clear it's a new object
    - No confusion about persistence
    - Immutability!

2. method that returns a copy?

   ```ruby
    proofing_job_result.as_done
    
    def as_done
      dup.tap do |job_result|
        job_result.result.symbolize_keys!
        job_result.pii.symbolize_keys!
        job_result.state = :done
      end
    end
   ```
    - Not clear from just the name it's a dupe, you have to look at the source :/

Reply:
`!` is Rubyism for modifying an object in place, but I will rarely turn down immutability 🙂